adapter/src/utils/localeUtils.js

```diff
@@ -0,0 +1,162 @@
+import i18n from '@dhis2/d2-i18n'
+import moment from 'moment'
+
+// Init i18n namespace
+const I18N_NAMESPACE = 'default'
+i18n.setDefaultNamespace(I18N_NAMESPACE)
+
+/**
+ * userSettings.keyUiLocale is expected to be formatted by Java's
+ * Locale.toString():
+ * https://docs.oracle.com/javase/8/docs/api/java/util/Locale.html#toString--
+ * We can assume there are no Variants or Extensions to locales used by DHIS2
+ * @param {Intl.Locale} locale
+ */
+const parseJavaLocale = (locale) => {
+    const [language, region, script] = locale.split('_')
+
+    let languageTag = language
+    if (script) {
+        languageTag += `-${script}`
+    }
+    if (region) {
+        languageTag += `-${region}`
+    }
+
+    return new Intl.Locale(languageTag)
+}
+
+/**
+ * @param {UserSettings} userSettings
+ * @returns Intl.Locale
+ */
+export const parseLocale = (userSettings) => {
+    try {
+        // proposed property
+        if (userSettings.keyUiLanguageTag) {
+            return new Intl.Locale(userSettings.keyUiLanguageTag)
+        }
```

Comment:
This is the new property Austin and I proposed, as a proof of concept of forward and backward compatibilty